autoinstallation/bin/agama-auto

```diff
@@ -1,35 +1,58 @@
-#!/usr/bin/sh
-set -ex
+#!/usr/bin/bash
+set -x
 
-# Temporarily skip the AutoYaST XML validation
-export YAST_SKIP_XML_VALIDATION=1
+# Find the URL given by the user.
+find_user_url() {
+  local url=$1
+  if [ -z "$1" ]; then
+    url=$(grep '\(agama\|inst\).auto=' </run/agama/cmdline.d/agama.conf | awk -F ':?(inst|agama).auto=' '{sub(/ .*$/, "", $2); print $2}')
+    echo $url
+  else
+    echo $1
+  fi
+}
 
-if [ -z "$1" ]; then
-  url=$(grep '\(agama\|inst\).auto=' < /run/agama/cmdline.d/agama.conf | awk -F ':?(inst|agama).auto=' '{sub(/ .*$/, "", $2); print $2}')
-else
-  url="$1"
-fi
+# Try to import a profile from a list of URLs
+import_profile() {
+  local url=$1
 
-if [ -z "$url" ]; then
-  echo "no autoinstallation profile"
-  exit 0
-fi
+  if [ -n "$url" ]; then
+    echo "Using the profile located at $url"
+    agama profile import "$url" && return 0
+  else
+    urls=("label://OEMDRV/autoinst.xml" "label://OEMDRV/autoinst.jsonnet" "label://OEMDRV/autoinst.json" "file:///autoinst.xml" "file:///autoinst.jsonnet" "file:///autoinst.json")
```

Comment:
Nitpicking: I'd probably change the order and search for Agama-specific "profiles" first, then AutoYaST.